### DISCUSSION TITLE
Rust nightly missing `backtrace = "^0.3.61"`, unable to build to the latest version of vaultwarden

### DISCUSSION THREAD

**Discussion by GNY-001F2** [dani-garcia/vaultwarden#2138] [Q&A]
I have a bitwarden_rs (probably v1.17.x, I don't remember) and I am trying to update it to the latest version.

Unfortunately, cargo is failing to build the program with 

```
error: failed to select a version for the requirement `backtrace = "^0.3.61"`
candidate versions found which didn't match: 0.3.55, 0.3.54, 0.3.53, ...
location searched: crates.io index
```

I have used rustup to update to the latest available nightly build of rust, and that has not helped either.

How do I work around this error?

**Comment by BlackDex** [comment]
See https://github.com/dani-garcia/vaultwarden/discussions/2056#discussioncomment-1513405

**Reply by GNY-001F2** [reply]
> See [#2056 (reply in thread)](https://github.com/dani-garcia/vaultwarden/discussions/2056#discussioncomment-1513405)

Thanks! Now I'm having trouble compiling the web vault, with npm failing to find the script `dist`

**Reply by BlackDex** [reply]
Check the Dockerfile, it needs a specific version of npm.
If I'm correct it needs node 14 and npm 7

**Reply by GNY-001F2** [reply]
> Check the Dockerfile, it needs a specific version of npm. If I'm correct it needs node 14 and npm 7

Actually I am building straight from git natively. I am on npm 8 if I am not mistaken. Should I download the older version to make it work?

EDIT: seems like npm14 installs npm 6.14.xxx while npm16 installs npm 8.1.0 on opensuse tumbleweed. there's no npm 7 any longer

**Reply by BlackDex** [reply]
I figured that you were not using the Dockerfile, but it has pointers in there which you need to follow. One of them is using npm7.

This is also mentioned on the Bitwarden web repo it self and visible in there build workflow.

**Reply by GNY-001F2** [reply]
Seems like npm14 installs npm 6.14.xxx while npm16 installs npm 8.1.0 on opensuse tumbleweed. there's no npm 7 any longer

**Reply by BlackDex** [reply]
Again, check the Dockerfile.
`npm -g install npm@7`

**Reply by GNY-001F2** [reply]
> Again, check the Dockerfile. `npm -g install npm@7`

I checked the dockerfile in docker/amd64 and I couldn't figure out how to force npm v7 to be used when building/installing the web vault.

I have tried npm -g install npm@7 with both npm14 (6.14) and npm16 (8.1) but npm -v still refers to the originally installed version.
